### PR TITLE
letsencrypt: support DuckDNS DNS challenge

### PR DESCRIPTION
**Pull request by iranl**
Adds DuckDNS DNS challenge support.

CNAME for custom domain "test.example.com" with DuckDNS domain "abc.duckdns.org" should be set up as follows:

```commandline
_acme-challenge.test.example.com. 600 IN CNAME abc.duckdns.org.
```

Note: Alias / custom domain support for obtaining a Let's Encrypt certificate in the DuckDNS addon is broken (#2505) and devs have proposed to remove support for aliases altogether (#2662 #2964).

Proposed solution was to use the DuckDNS addon for keeping a dynamic IP up to date only and configure the Let's Encrypt addon to obtain a certificate (through a HTTP challenge or third party DNS provider that is supported by the Let's Encrypt addon).

This PR allows for DuckDNS DNS challenges including for custom domains and as such no ports need to be opened to allow a HTTP challenge and no secondary DNS provider is needed to complete the DNS challenge for custom domains.

### CODE REVIEW

File: letsencrypt/build.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ args:
   CRYPTOGRAPHY_VERSION: 36.0.2
   CERTBOT_DNS_AZURE_VERSION: 1.5.0
   CERTBOT_DNS_DIRECTADMIN_VERSION: 0.0.24
+  CERTBOT_DNS_DUCKDNS_VERSION: v1.3
```
Comment (agners) [inline letsencrypt/build.yaml:15]: It seems that `v` isn't hurting here, but for consistency I'll drop it.

```suggestion
  CERTBOT_DNS_DUCKDNS_VERSION: 1.3
```

**Review by agners — APPROVED** [review]

**Comment by iranl** [comment]
Tnx for reviewing @agners.

Build on armhf and armv7 seems to fail on building cryptography from source. 
A binary wheel is available and being used on all other platforms.

The information on [https://cryptography.io/en/latest/installation/#alpine](https://cryptography.io/en/latest/installation/#alpine) suggests that 

> The Rust available by default in Alpine < 3.17 is older than the minimum supported version.

Merging #3325 with its update to Alpine 3.18 could solve this issue?

**Comment by agners** [comment]
> Merging #3325 with its update to Alpine 3.18 could solve this issue?

Yeah it should. #3325 builds fine it seems. So best is to rebase this once #3325 got merged.

**Comment by agners** [comment]
Ok, #3325 is merged now, conflicts need to be resolved now and version in confiig/CHANGELOG adjusted.

File: letsencrypt/config.yaml [hunk 2]
```diff
@@ -72,7 +73,7 @@ schema:
     ovh_endpoint: str?
     propagation_seconds: int(60,3600)?
     provider: "list(dns-azure|dns-cloudflare|dns-digitalocean|\
-      dns-directadmin|dns-dnsimple|dns-dnsmadeeasy|dns-gehirn|\
+      dns-directadmin|dns-dnsimple|dns-dnsmadeeasy|dns-duckdns|dns-gehirn|\
       dns-google|dns-google-domains|\
```
Comment (agners) [inline letsencrypt/config.yaml:77]:
```suggestion
      dns-directadmin|dns-dnsimple|dns-dnsmadeeasy|dns-duckdns|\
      dns-gehirn|dns-google|dns-google-domains|\
```

**Review by agners — APPROVED** [review]
Small nit, but other than that LGTM.

**Pull request merged**